projects/gltf_basic_materials/GltfBasicMaterials.cpp

```diff
@@ -0,0 +1,339 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "GltfBasicMaterials.h"
+#include "ppx/scene/scene_gltf_loader.h"
+#include "ppx/graphics_util.h"
+
+using namespace ppx;
+
+#if defined(USE_DX12)
+const grfx::Api kApi = grfx::API_DX_12_0;
+#elif defined(USE_VK)
+const grfx::Api kApi = grfx::API_VK_1_1;
+#endif
+
+void GltfBasicMaterialsApp::Config(ppx::ApplicationSettings& settings)
+{
+    settings.appName                    = "gltf_basic_materials";
+    settings.enableImGui                = true;
+    settings.grfx.api                   = kApi;
+    settings.grfx.enableDebug           = true;
+    settings.window.width               = 1920 * 1;
+    settings.window.height              = 1080 * 1;
+    settings.window.resizable           = false;
+    settings.grfx.swapchain.depthFormat = grfx::FORMAT_D32_FLOAT;
+    settings.allowThirdPartyAssets      = true;
+}
+
+void GltfBasicMaterialsApp::Setup()
+{
+    // Per frame data
+    {
+        PerFrame frame = {};
+
+        PPX_CHECKED_CALL(GetGraphicsQueue()->CreateCommandBuffer(&frame.cmd));
+
+        grfx::SemaphoreCreateInfo semaCreateInfo = {};
+        PPX_CHECKED_CALL(GetDevice()->CreateSemaphore(&semaCreateInfo, &frame.imageAcquiredSemaphore));
+
+        grfx::FenceCreateInfo fenceCreateInfo = {};
+        PPX_CHECKED_CALL(GetDevice()->CreateFence(&fenceCreateInfo, &frame.imageAcquiredFence));
+
+        PPX_CHECKED_CALL(GetDevice()->CreateSemaphore(&semaCreateInfo, &frame.renderCompleteSemaphore));
+
+        fenceCreateInfo = {true}; // Create signaled
+        PPX_CHECKED_CALL(GetDevice()->CreateFence(&fenceCreateInfo, &frame.renderCompleteFence));
+
+        mPerFrame.push_back(frame);
+    }
+
+    // Load GLTF scene
+    {
+        scene::GltfLoader* pLoader = nullptr;
+        //
+        PPX_CHECKED_CALL(scene::GltfLoader::Create(
+            GetAssetPath("scene_renderer/scenes/tests/gltf_test_basic_materials.glb"),
+            // GetAssetPath("scene_renderer/scenes/tests/gltf_test_materials.gltf"),
+            nullptr,
+            &pLoader));
+
+        PPX_CHECKED_CALL(pLoader->LoadScene(GetDevice(), 0, &mScene));
+        PPX_ASSERT_MSG((mScene->GetCameraNodeCount() > 0), "scene doesn't have camera nodes");
+        PPX_ASSERT_MSG((mScene->GetMeshNodeCount() > 0), "scene doesn't have mesh nodes");
+
+        delete pLoader;
+    }
+
+    // IBL Textures
+    {
+        PPX_CHECKED_CALL(grfx_util::CreateIBLTexturesFromFile(
+            GetDevice()->GetGraphicsQueue(),
+            GetAssetPath("poly_haven/ibl/old_depot_4k.ibl"),
+            // GetAssetPath("poly_haven/ibl/abandoned_workshop_02_4k.ibl"),
```

Comment:
Should probably remove this line

Reply:
And remove the new assets since they seem to be unused.